Raven.Database/Prefetching/PrefetchingBehavior.cs

```diff
@@ -1498,16 +1502,19 @@ public void CleanupDocuments(Etag lastIndexedEtag)
                 if (docToRemove.Value.All(etag => lastIndexedEtag.CompareTo(etag) > 0) == false)
                     continue;
 
-                HashSet<Etag> _;
-                documentsToRemove.TryRemove(docToRemove.Key, out _);
+                ImmutableAppendOnlyList<Etag> value;
+                if(documentsToRemove.TryRemove(docToRemove.Key, out value))
+                {
+                    currentlyTrackedDeletedItems -= value.Count; //doesn't have to be accurate
```

Comment:
Doesn't have to be accurate, but this can cause large gaps.
Use `Interlocked` here, please

Reply:
no it won't since we zero it...

Reply:
Nevertheless